yt_dlp/extractor/agalega.py

```diff
@@ -0,0 +1,336 @@
+
+    def _get_access_token(self, video_id: str) -> dict[str, str] | None:
+        token_url = 'https://www.agalega.gal/api/fetch-api/jwt/token'
+        body_data = json.dumps({
+            'username': None,
+            'password': None,
+            'client': 'crtvg',
+            'checkExistsCookies': False,
+        }).encode()
+
+        token_raw = self._download_json(token_url, video_id, data=body_data)
+
+        if token_raw is not None:
+            access_token = str_or_none(token_raw.get('access'))
+            if access_token is not None:
+                return {'authorization': f'jwtok {access_token}'}
+            else:
+                return None
+        return None
+
+
+class AGalegaVideosIE(AGalegaBaseIE):
+    IE_NAME = 'agalega:videos'
+    _VALID_URL = r'https?://(?:www\.)?agalega\.gal/videos/(?:detail/)?(?P<id>[0-9]+)'
+    _TESTS = [{
+        'url': 'https://www.agalega.gal/videos/288664-lr-ninguencheconta',
+        'md5': '04533a66c5f863d08dd9724b11d1c223',
+        'info_dict': {
+            'id': '288664',
+            'title': 'Roberto e Ángel Martín atenden consultas dos espectadores',
+            'description': 'O cómico ademais fai un repaso dalgúns momentos da súa traxectoria profesional',
+            'thumbnail': 'https://crtvg-bucket.flumotion.cloud/content_cards/2ef32c3b9f6249d9868fd8f11d389d3d.png',
+            'ext': 'mp4',
+        },
+    },
+        {
+        'url': 'https://www.agalega.gal/videos/detail/296152-pulso-activo-7',
+        'md5': '26df7fdcf859f38ad92d837279d6b56d',
+        'info_dict': {
+            'id': '296152',
+            'title': 'Pulso activo | 18-11-2025',
+            'description': 'Anxo, Noemí, Silvia e Estrella  comparten as sensacións da clase de Eddy.',
+            'thumbnail': 'https://crtvg-bucket.flumotion.cloud/content_cards/a6bb7da6c8994b82bf961ac6cad1707b.png',
+            'ext': 'mp4',
+        },
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        info_url = f'https://api-agalega.interactvty.com/api/2.0/contents/content/{video_id}/?optional_fields=image,is_premium,short_description,has_subtitle'
+        media_url = f'https://api-agalega.interactvty.com/api/2.0/contents/content_resources/{video_id}/?optional_fields=media_url'
+        auth_header = self._get_access_token(video_id)
+
+        if auth_header is not None:
+            info_raw = self._download_json(info_url, video_id, headers=auth_header)
+            media_raw = self._download_json(media_url, video_id, headers=auth_header)
+
+            if info_raw is not None and media_raw is not None:
+                m3u8_url = traverse_obj(media_raw, ('results', ..., 'media_url', {url_or_none}, any, {require('media_url')}))
+                formats = self._extract_m3u8_formats_and_subtitles(
+                    m3u8_url, video_id, ext='mp4',
+                    m3u8_id='hls', entry_protocol='m3u8_native', fatal=False,
+                )
```

Comment:
Same here, we should also extract all possible media URLs and not silently discard subtitles.
```suggestion
        formats = []
        subtitles = {}
        for m3u8_url in traverse_obj(resource_data, ('results', ..., 'media_url', {url_or_none})):
            fmts, subs = self._extract_m3u8_formats_and_subtitles(
                m3u8_url, video_id, ext='mp4', m3u8_id='hls')
            formats.extend(fmts)
            self._merge_subtitles(subs, target=subtitles)
```